graphql-database-manager-dynamo/src/main/java/com/fleetpin/graphql/database/manager/dynamo/DynamoDb.java

```diff
@@ -815,24 +815,44 @@ private CompletableFuture<List<DynamoItem>> query(String organisationId, String
 		var id = keys.get("id");
 		Map<String, AttributeValue> keyConditions = new HashMap<>();
 		keyConditions.put(":organisationId", organisationIdAttribute);
-		if (id != null && !id.s().isEmpty()) {
-			keyConditions.put(":table", id);
+
+		String index = query.getIndex();
+		boolean consistentRead = true;
+
+		if (id != null && !id.s().trim().isEmpty()) {
+            index = null;
+            keyConditions.put(":table", id);
+		} else if (query.getThreadIndex() != null && query.getThreadCount() != null) {
```

Comment:
would do this check first and if id is not null error so users know its not fired and need to update logic